1_fetch/src/get_wqp_inventory.R

```diff
@@ -27,23 +28,128 @@ inventory_wqp <- function(grid, char_names, ...){
+                        crs_out == "WGS84" ~ 4326)
+  
+  # Transform sites into a consistent crs defined by epsg_out
+  sites_transformed <- sites %>%
+    # transform the sites data frame into a list where all sites within each 
+    # list element share the same crs/datum
+    rename(datum = HorizontalCoordinateReferenceSystemDatumName) %>%
+    split(.,.$datum) %>%
+    # Each list element contains sites with different crs. Transform the sites 
+    # within each element to the desired crs
+    lapply(., function(x){
+      # assume unknown crs (datum equals "UNKNWN", "OTHER") correspond with WGS84
+      epsg_in <- case_when(x$datum[1] == "NAD83" ~ 4269,
+                           x$datum[1] == "WGS84" ~ 4326,
+                           x$datum[1] == "NAD27" ~ 4267,
+                           x$datum[1] == "UNKWN" ~ 4326,
+                           x$datum[1] == "OTHER" ~ 4326)
+      
+      # convert data frame to sf object and transform to desired crs
+      if(!is.na(epsg_in)){
+        sites_transformed <- x %>%
+          sf::st_as_sf(coords=c("lon","lat"), crs = epsg_in) %>%
+          sf::st_transform(epsg_out) %>%
+          mutate(lon_new = sf::st_coordinates(.)[,1],
+                 lat_new = sf::st_coordinates(.)[,2],
+                 datum_new = crs_out) %>%
+          sf::st_drop_geometry() %>%
+          select(-datum) %>%
+          rename("lon"="lon_new", "lat"="lat_new", "datum"="datum_new")
+      } else {
+        sites_transformed <- x
+        message(sprintf("Unable to transform sites with datum = %s; returning sites untransformed",
+                        x$datum[1]))
+      }
+      return(sites_transformed)
+    }) %>%
+    # combine transformed sites into a single data frame
+    bind_rows()
+  
+  return(sites_transformed)
+  
+}
+
+
+
+#' Subset WQP inventory by the area of interest polygon
+#' 
+#' @description Function to filter out any sites in the WQP inventory that are
+#' outside of the area of interest
+#' 
+#' @param wqp_inventory data frame containing sites returned from the WQP query; 
+#' contains columns lon, lat, and HorizontalCoordinateReferenceSystemDatumName.
+#' @param aoi_sf sf object representing the area of interest
+#' 
+#' @value returns a data frame containing sites from the Water Quality Portal that 
+#' are located within the area of interest.
+#' 
+subset_inventory <- function(wqp_inventory, aoi_sf){
+  
+  # Harmonize different coordinate reference systems used across sites
+  queried_sites_transformed <- transform_site_locations(wqp_inventory, crs_out= "WGS84") %>%
+    sf::st_as_sf(coords = c("lon","lat"), crs = 4326) 
+  
+  # Filter wqp inventory to only include sites within area of interest + 100 m buffer 
+  # to ensure we retain all sites within the AOI.
+  queried_sites_aoi <- queried_sites_transformed %>%
+    sf::st_filter(y = sf::st_transform(aoi_sf,sf::st_crs(.)),
+                  .predicate = st_is_within_distance,dist=units::set_units(100, m)) %>%
```

Comment:
Is 100 m something that should be parameterizable?

Reply:
Thanks! Yes, this value should be parameterizable. I've added `buffer_dist_m` as an argument to this function and also added some documentation to indicate how we're using it.